app/Components/chatRoom/chatRoom.js

```diff
@@ -0,0 +1,430 @@
+import React, {Component} from 'react';
+import io from 'socket.io-client';
+import {connect} from 'react-redux';
+import EmojiSelector, { Categories } from 'react-native-emoji-selector';
+import {
+  ScrollView,
+  Text,
+  StyleSheet,
+  View,
+  Pressable,
+  Image,
+  TextInput,
+  TouchableOpacity
+} from 'react-native';
+import MenuButton from '../../../assests/horizontalDots.png';
+import MessagePop from './messagePop';
+import EmojiAdd from '../../../assests/emoji-add-icon.png'
+
+class ChatRoom extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      chatMessage: '',
+      message: '',
+      messages: [],
+      isOponentTyping: false,
+      showEmoji:false
+      // MessagePopUp:false
+    };
+  }
+  socket = null;
+  componentDidMount = () => {
+    this.socket = io('https://ptchatindia.herokuapp.com/', {
+      transports: ['websocket'],
+    });
+    this.socket.emit('joinRoom', {
+      username: this.props.user.username,
+      client2: this.props.client.username,
+    });
+    this.socket.once('messages', this.onMessages);
+    this.socket.on('message', this.onMessage);
+    this.socket.on('typing-start', this.onTyping);
+    this.socket.on('typing-end', this.onTyping);
+  };
+
+  componentWillUnmount() {
+    this.socket.off('message', this.onMessage);
+    this.socket.off('messages', this.onMessages);
+    this.socket.off('typing-start', this.onTyping);
+    this.socket.off('typing-end', this.onTyping);
+  }
+
+  onTyping = data => {
+    if (this.props.user.username !== data.username) {
+      this.setState({isOponentTyping: data.typing});
+    }
+  };
+
+  onMessage = data => {
+    let messages = this.state.messages;
+    Object.assign(data, {messagePopUp: false});
+    messages.push(data);
+    this.previousDate = null;
+    this.setState({messages: messages});
+  };
+
+  onMessages = data => {
+    this.setState({messages: data.messages});
+    this.socket.off('messages', true);
+  };
+
+  sendTypingStartStatus = () => {
+    this.socket.emit('typing-start', {
+      username: this.props.user.username,
+      client2: this.props.client.username,
+    });
+  };
+
+  sendTypingEndStatus = () => {
+    this.socket.emit('typing-end', {
+      username: this.props.user.username,
+      client2: this.props.client.username,
+    });
+  };
+
+  send = () => {
+      console.log('send');
+    if (this.state.chatMessage) {
+      this.socket.emit('chat', {
+        username: this.props.user.username,
+        client2: this.props.client.username,
+        message: this.state.chatMessage,
+      });
+      this.setState({chatMessage: ""});
```

Comment:
Give a comment there why you are doing empty to that chatMessage variable.Try to write comments Before starting the functions